src/main.rs

```diff
@@ -237,6 +266,13 @@ fn main() -> Result<()> {
         }
         info!(status = "done", "kubernetes poller bootstrap");
 
+        // Spawn the tokio task used by the CallbackHandler
+        let callback_handle = tokio::spawn(async move {
+            info!(status = "init", "CallbackHandler task");
+            callback_handler.loop_eval().await;
```

Comment:
`loop_eval()` does an infinite loop: https://github.com/kubewarden/policy-evaluator/pull/67/files#diff-26f1a863d124e9b03db590e7ab50f50548b71b6e91fb40d0c2e510565743fb8cR104-R131. As far as I understand, `tokio::spawn()` **could** run on the same thread (from the docs: "The spawned task may execute on the current thread, or it may be sent to a different thread to be executed").

if it is executed in the current thread I fear we might never make progress on futures if we are stuck on an endless loop in our side.

Maybe this comes from not fully understanding the runtime properties of Tokio & futures, could this happen?

Reply:
I have the same feeling here, thanks for spotting it Rafa. Maybe the blocking future should be in its own `thread::spawn`, or use `tokio::spawn_blocking`? (https://docs.rs/tokio/1.14.0/tokio/task/fn.spawn_blocking.html)

Reply:
this is not a problem, I found the real root cause of the performance issue: https://github.com/kubewarden/policy-evaluator/pull/67#issuecomment-984471982